internal/core/url.go

```diff
@@ -36,12 +36,26 @@ func queryEncodeWithoutEmptyValues(v nurl.Values) string {
 	return buf.String()
 }
 
+// Parse parses a URL. If it fails, it tries again with "https://" prefix.
+func Parse(url string) (*nurl.URL, error) {
+	urlParsed, err := nurl.Parse(url)
+	if err != nil || urlParsed.Scheme == "" || urlParsed.Hostname() == "" {
+		if strings.HasPrefix(url, "https://") {
+			return nil, fmt.Errorf("URL is not valid")
+		}
+
+		return Parse("https://" + url)
+	}
+
+	return urlParsed, err
+}
+
 // RemoveUTMParams removes the UTM parameters from URL.
 func RemoveUTMParams(url string) (string, error) {
 	// Parse string URL
-	tmp, err := nurl.Parse(url)
-	if err != nil || tmp.Scheme == "" || tmp.Hostname() == "" {
-		return url, fmt.Errorf("URL is not valid")
+	tmp, err := Parse(url)
+	if err != nil {
+		return url, err
```

Comment:
Let's make errors traceable now that we are on it

```suggestion
		return url, fmt.Errorf("error removing utm params: %w", err)
```